domain/application/service/application.go

```diff
@@ -1336,6 +1356,21 @@ func (s *Service) GetApplicationConstraints(ctx context.Context, appID coreappli
 	return constraints.EncodeConstraints(cons), errors.Capture(err)
 }
 
+// GetApplicationEndpointBindings returns the mapping for each endpoint name and
+// the space ID it is bound to (or empty if unspecified). When no bindings are
+// stored for the application, defaults are returned.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (s *Service) GetApplicationEndpointBindings(ctx context.Context, appID coreapplication.ID) (map[string]string, error) {
+	if err := appID.Validate(); err != nil {
+		return nil, errors.Errorf("application ID: %w", err)
```

Comment:
nit: `validating application id: %w`.

We try our best to make errors start with the verb that was the genesis of the error.